include/zephyr/bluetooth/classic/avdtp.h

```diff
@@ -4,49 +4,154 @@
+	/** The requested command has an incorrect payload format */
+	BT_AVDTP_BAD_PAYLOAD_FORMAT = 0x18,
+	/** The requested command is not supported by the device */
+	BT_AVDTP_NOT_SUPPORTED_COMMAND = 0x19,
+	/** The reconfigure command is an attempt to reconfigure a transport service codec_cap
+	 * of the SEP. Reconfigure is only permitted for application service codec_cap
+	 */
+	BT_AVDTP_INVALID_CAPABILITIES = 0x1A,
+	/** The requested Recovery Type is not defined in AVDTP */
+	BT_AVDTP_BAD_RECOVERY_TYPE = 0x22,
+	/** The format of Media Transport Capability is not correct */
+	BT_AVDTP_BAD_MEDIA_TRANSPORT_FORMAT = 0x23,
+	/** The format of Recovery Service Capability is not correct */
+	BT_AVDTP_BAD_RECOVERY_FORMAT = 0x25,
+	/** The format of Header Compression Service Capability is not correct */
+	BT_AVDTP_BAD_ROHC_FORMAT = 0x26,
+	/** The format of Content Protection Service Capability is not correct */
+	BT_AVDTP_BAD_CP_FORMAT = 0x27,
+	/** The format of Multiplexing Service Capability is not correct */
+	BT_AVDTP_BAD_MULTIPLEXING_FORMAT = 0x28,
+	/** Configuration not supported */
+	BT_AVDTP_UNSUPPORTED_CONFIGURAION = 0x29,
+	/** Indicates that the ACP state machine is in an invalid state in order to process the
+	 * signal. This also includes the situation when an INT receives a request for the
+	 * same command that it is currently expecting a response
+	 */
+	BT_AVDTP_BAD_STATE = 0x31,
+};
+
+/** @brief Stream End Point Role */
+enum bt_avdtp_role_type {
+	/** Source Role */
+	BT_AVDTP_SOURCE = 0,
+	/** Sink Role */
+	BT_AVDTP_SINK = 1
+};
+
+/** @brief Stream End Point Media Type */
+enum bt_avdtp_media_type {
+	/** Audio Media Type */
+	BT_AVDTP_AUDIO = 0x00,
+	/** Video Media Type */
+	BT_AVDTP_VIDEO = 0x01,
+	/** Multimedia Media Type */
+	BT_AVDTP_MULTIMEDIA = 0x02
+};
+
+/** @brief AVDTP SEID Information AVDTP_SPEC V13 Table 8.8 */
 struct bt_avdtp_seid_info {
-	/** Stream End Point ID */
+#ifdef CONFIG_LITTLE_ENDIAN
+	/** Reserved */
+	uint8_t rfa0:1;
+	/** End Point usage status */
+	uint8_t inuse:1;
+	/** Stream End Point ID that is the identifier of the stream endpoint */
+	uint8_t id:6;
+	/** Reserved */
+	uint8_t rfa1:3;
+	/** This field indicates if the stream end-point is SNK or SRC, @ref bt_avdtp_role_type */
+	uint8_t tsep:1;
+	/** Media-type of the End Point, @ref bt_avdtp_media_type
+	 * Only @ref BT_AVDTP_AUDIO is supported now.
+	 */
+	uint8_t media_type:4;
+#else
+	/** Stream End Point ID that is the identifier of the stream endpoint */
 	uint8_t id:6;
 	/** End Point usage status */
 	uint8_t inuse:1;
 	/** Reserved */
 	uint8_t rfa0:1;
-	/** Media-type of the End Point */
+	/** Media-type of the End Point, @ref bt_avdtp_media_type
+	 * Only @ref BT_AVDTP_AUDIO is supported now.
+	 */
 	uint8_t media_type:4;
-	/** TSEP of the End Point */
+	/** This field indicates if the stream end-point is SNK or SRC, @ref bt_avdtp_role_type */
 	uint8_t tsep:1;
 	/** Reserved */
 	uint8_t rfa1:3;
+#endif
```

Comment:
Same here regarding the duplicated documentation. Sorry about the back-and-forth with bitfields vs accessor macros solutions, but now that I see the end result the accessor macros (especially with Zephyr's `FIELD_GET()`, `FIELD_PREP()`, `GENMASK()`, etc helpers) feels like a cleaner approach. Feel free to provide counterarguments of course (e.g. if the documentation can be moved into a single place, although then it wouldn't be inline anymore), but the macro-based option is what seems like a slightly better solution to me right now.

Reply:
Codes are updated. The bit-fields structure is only used internally.